### PR TITLE
Logger interface extracted from yii\log\Logger

### PR DESCRIPTION
**Pull request by njasm**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | no
| Fixed issues  | #14997

### CODE REVIEW

**Review by samdark — APPROVED** [review]

File: framework/log/LoggerInterface.php [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+namespace yii\log;
+
+/**
+ * LoggerInterface defines a Logger that records logged messages in memory and sends them to different targets.
+ *
+ * A Logger instance can be accessed via `Yii::getLogger()`.
+ * For more details and usage information on Logger, see the [guide article on logging](guide:runtime-logging).
+ *
+ * @author Nelson J Morais <njmorais@gmail.com>
```
Comment (SilverFire) [inline framework/log/LoggerInterface.php]: API was originally authored by Qiang. I think his authoring copyright should not be removed.

**Review by SilverFire — CHANGES_REQUESTED** [review]

**Comment by SilverFire** [comment]
Good otherwise.

**Review by dmirogin — APPROVED** [review]

**Comment by klimov-paul** [comment]
Such changeset is absolutely redundant since #14611

File: framework/log/LoggerInterface.php (hunk 1 above)
Comment (njasm) [reply framework/log/LoggerInterface.php]: Done. 👍

**Comment by samdark** [comment]
Closing because of what @klimov-paul pointed out.

**Comment by samdark** [comment]
Thank you, @njasm for doing it though and sorry your commits aren't merged.

**Comment by njasm** [comment]
No problem, makes sense :+1

**Pull request closed without merging**